apps/src/applab/designElements/ThemeDropdown.jsx

```diff
@@ -0,0 +1,84 @@
+// Dropdown for theme selection that shows sample icons of what
+// each theme will look like
+import PropTypes from 'prop-types';
+import React from 'react';
+import {themeOptionsForSelect} from '../constants';
+import Select from 'react-select';
+import 'react-select/dist/react-select.css';
+
+export default class ThemeDropdown extends React.Component {
+  static propTypes = {
+    initialValue: PropTypes.string.isRequired,
+    handleChange: PropTypes.func.isRequired,
+    desc: PropTypes.node
+  };
+
+  state = {
+    selectedValue: this.props.initialValue
+  };
+  handleChange = event => {
+    this.props.handleChange(event.value);
+    this.setState({selectedValue: event.value});
+  };
+
+  // Need to reset theme value if screen switches
+  componentWillReceiveProps(nextProps) {
+    const {initialValue} = nextProps;
+    if (this.props.initialValue !== initialValue) {
+      this.setState({selectedValue: initialValue});
+    }
+  }
+
+  // need map of value-> {icon, display name}
+  render() {
+    const styles = {
```

Comment:
NIT: move the styles outside the component (i.e. like [this](https://github.com/code-dot-org/code-dot-org/blob/a425487209c5d437c56716d21c87f2fb35118f2a/apps/src/code-studio/components/libraries/LibraryListItem.jsx#L10))